Add robotframework-clang

maxrossello opened the pull request:
Adding recipe for robotframework-clang library.

Commit:
Add robotframework-clang recipe

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/robotframework-clang/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/robotframework-clang/meta.yaml**:

 * ❌ `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`, or selector on line 31:
    - libcxx  # [linux or osx]


For **recipes/robotframework-clang/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20574332484](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20574332484). Examine the logs at this URL for more detail.</sub>

Commit:
Fix: remove noarch: python

Commit:
Fix: removed python requirement

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/robotframework-clang/meta.yaml```) and found it was in an excellent condition.

Commit:
fix: correct conda selector syntax in recipe

Commit:
fix: hardcode version in conda selector for compatibility

Commit:
fix: robust path discovery for CI/Docker environments

    - Pass include paths directly via extra_arguments to ensure headers are found in isolated containers.
    - Dynamic JUPYTER_PATH setup to locate kernels on Windows.
    - Cleaned up pyproject.toml license metadata.

Commit:
fix: linux in docker with needed packages, windows path references to missing kernel

maxrossello committed:
Merge branch 'main' into robotframework-clang

Commit:
Add robotframework-clang recipe

Commit:
Fixes for multiarch

Commit:
Fixes for multiarch

Commit:
Fixed compilers specs

Commit:
Conda config assessments

Commit:
Fixes

Commit:
Fixes

Commit:
Fixes

Commit:
Fixes

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/robotframework-clang/meta.yaml```) and found it was in an excellent condition.

Commit:
Fixes

Commit:
Fixes

maxrossello committed:
Merge branch 'main' into robotframework-clang

Commit:
Fix: Address Windows build and linking issues

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/robotframework-clang/meta.yaml```) and found it was in an excellent condition.

maxrossello committed:
Merge branch 'main' into robotframework-clang

maxrossello committed:
Merge branch 'main' into robotframework-clang

Commit:
Fix Windows build: add MSVC runtime linking, update recipe dependencies and fix test compilation

Commit:
Moving to pixi build envs

Commit:
Complying to conda linter's new requirements

Commit:
rf-clang-0.2.1
Fixes for python 3.10
Fixes for osx dependencies

Commit:
Improved windows resiliency

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/robotframework-clang/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/robotframework-clang/recipe.yaml**:

 * ❌ Selectors in comment form no longer work in v1 recipes. Instead, if / then / else maps must be used. See lines [23, 24, 38, 50, 51].
 * ❌ Non noarch packages should have python requirement without any version constraints.
 * ❌ Non noarch packages should have python requirement without any version constraints.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/21562472600](https://github.com/conda-forge/conda-forge-webservices/actions/runs/21562472600). Examine the logs at this URL for more detail.</sub>

Commit:
Further recipe assessments

Commit:
Revert python versioning

Commit:
Bump version to 0.2.5 and improve Windows/Linux JIT resilience

Commit:
Fixed again for python 3.10

Commit:
Further f-strings fixed

Commit:
Including headers one at a time for debug purposes on windows

Commit:
Fixes for Windows

Commit:
Using clang 21 and final fixes

Commit:
Fixed sha reference

Commit:
Version 1.0.0

Commit:
Version 1.0.1
Minor integration and assessment of documentation

Commit:
Fixed the pkg tag

maxrossello committed:
Merge branch 'main' into robotframework-clang

h-vetinari commented:
> Or am I missing something?

Yes; stdlib is necessary (and will be populated correctly so that the respective metapackage can be resolved for conda-forge recipes). Daniel [described](https://github.com/conda-forge/staged-recipes/pull/31788#discussion_r2928554238) how to do this via `variants.yaml` (or similarly via `conda_build_config.yaml`)

Commit:
Version 1.0.3
Improved conda recipe

Commit:
Merge branch 'robotframework-clang' of github.com:maxrossello/staged-recipes into robotframework-clang

Commit:
Fix typo in version placeholder

Commit:
Fix for osx-64

Commit:
Further fix for osx-64

Commit:
Further fix for osx-64

Commit:
Further fix for osx-64

Commit:
Rerun build

Commit:
Garbage collect in windows for tests keeping some file busy

Commit:
Garbage collect in windows for tests keeping some file busy

Commit:
Try to add a sleep at the end of windows tests for freeing up resources before rattler-build complains

Commit:
Try to add a sleep at the end of windows tests for freeing up resources before rattler-build complains

Commit:
Try to add a sleep at the end of windows tests for freeing up resources before rattler-build complains

maxrossello committed:
Merge branch 'main' into robotframework-clang

Commit:
Increase sleep after tests in windows

Commit:
Try even longer sleep

Commit:
Probing whether the problem with windows test section is not our

Commit:
Further test

Commit:
Fix for win staged-recipes

Commit:
v1.1.0

Commit:
Try fix for osx

Commit:
Reverting unintended change in pixi.toml

maxrossello commented:
@danielnachun I applied a tradeoff by adding clangxx explicitly only for osx and only in the test section.
If I don't do so (maybe because I require clangxx 21?) tests fail for not finding std::isnan. Looks like the macro ${{ compiler('cxx') }} does not set all the include paths properly.
Please let me know if this setup is acceptable
